pkg/storage/etcd3/compact_test.go

```diff
@@ -46,3 +46,41 @@ func TestCompact(t *testing.T) {
 		t.Errorf("Expecting ErrCompacted, but get=%v", err)
 	}
 }
+
+func TestCompactConflict(t *testing.T) {
```

Comment:
Could we add a test w/ multiple go-routines to see what happens with many api-servers.